daemon/containerd/image_list.go

```diff
@@ -35,3 +45,19 @@ func (i *ImageService) Images(ctx context.Context, opts types.ImageListOptions)
 
 	return ret, nil
 }
+
+func computeVirtualSize(ctx context.Context, image containerd.Image, snapshotter snapshots.Snapshotter) (int64, error) {
+	var virtualSize int64
+	diffIDs, err := image.RootFS(ctx)
+	if err != nil {
+		return virtualSize, err
+	}
+	for _, chainID := range identity.ChainIDs(diffIDs) {
+		usage, err := snapshotter.Usage(ctx, chainID.String())
```

Comment:
FWIW; some optimisations coming when upstreaming https://github.com/rumpl/moby/pull/17 - let me have a look if I can include that one in this PR

Reply:
Actually, that makes things a bit more complicated; let me keep it separate for now